### ISSUE TITLE
Doesn't work on MacOS Mojave 10.14.2 

### ISSUE THREAD

**Issue by vbratok** [journeyapps/node-sqlcipher#16]
I've been struggling to make the @journeyapps/sqlcipher work on a mac without any success for a few days already. 

I successfully installed `brew install openssl`
Used both `yarn` and `npm` installation. The installation part is done successfully.

However, when I run the example code from readme file using `node` I'm getting the following error 

> Vitaliis-MacBook-Pro:cipher-test vbratok$ node index.js
> events.js:173
>       throw er; // Unhandled 'error' event
>       ^
> 
> Error: SQLITE_ERROR: SQL logic error
> Emitted 'error' event at:
> 
> Vitaliis-MacBook-Pro:cipher-test vbratok$

That's it and nothing more. 

The test.db file is created in the folder, but it's empty and has zero bytes.
If I comment out the `db.run("PRAGMA key = 'mysecret'");` line, it works file, but db isn't encoded.

I did a lot of experiments and I checked that `node-sqlite3` and `SQLCipher` work fine separately and together when I bind them together as described [here](https://github.com/mapbox/node-sqlite3#on-os-x-with-homebrew) 

p.s. I noticed that there're a new version of openssl came out since the latest release of node-sqlcipher https://www.openssl.org/source/. Could it be an issue?

**Comment by cokuna-pavelkosolapov** [comment]
have the same Issue. It worked fine, but not anymore. Unfortunately, i'm not be able to say, when and why it was started. But now it doesn't work if "PRAGMA KEY" is using. Without it works.

here my npm install report
[report.txt](https://github.com/journeyapps/node-sqlcipher/files/2790935/report.txt)

I think the main problem is like this:
`ld: warning: object file (Release/obj.target/node_sqlite3/src/node_sqlite3.o) was built for newer OSX version (10.10) than being linked (10.7)` 
and this
`ld: warning: object file (/usr/local/opt/openssl/lib/libcrypto.a(evp_enc.o)) was built for newer OSX version (10.14) than being linked (10.7)` and lot of others same warnings

it crashed by trying to create a table with follow error
`{ [Error: SQLITE_ERROR: SQL logic error] errno: 1, code: 'SQLITE_ERROR' }`

if its important:
```
$ /usr/local/opt/openssl/bin/openssl version -a
OpenSSL 1.0.2q  20 Nov 2018
built on: reproducible build, date unspecified
platform: darwin64-x86_64-cc
options:  bn(64,64) rc4(ptr,int) des(idx,cisc,16,int) idea(int) blowfish(idx) 
compiler: clang -I. -I.. -I../include  -fPIC -fno-common -DOPENSSL_PIC -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -arch x86_64 -O3 -DL_ENDIAN -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -DECP_NISTZ256_ASM
OPENSSLDIR: "/usr/local/etc/openssl"```

**Comment by cokuna-pavelkosolapov** [comment]
an update. It really strange, but it works under electron, but not with node!

**Comment by andreddosantos** [comment]
Same issue here!

**Comment by vladanm** [comment]
Same issue here!

**Comment by rkistner** [comment]
Appears to have been an issue with openssl 1.0 on macOS, which seems solved with openssl 1.1.

Version 4.0.0 have now been released, with pre-built binaries for Mac based on openssl 1.1.
